src/node/history.h

```diff
@@ -559,7 +571,12 @@ namespace ccf
     void compact(kv::Version v) override
     {
       flush_pending();
-      replicated_state_tree.flush(v);
+      // Receipts can only be retrieved to the flushed index. Keep a range of
+      // history so that a range of receipts are available.
+      if (v > MAX_HISTORY_LEN)
+      {
+        replicated_state_tree.flush(v - MAX_HISTORY_LEN);
+      }
```

Comment:
I think this is going to run into issues with PBFT. Should be ok for Raft however.